vulnerabilities/importers/almalinux.py

```diff
@@ -47,3 +67,117 @@ def advisory_data(self) -> Iterable[AdvisoryData]:
+    aliases = raw_data.get("aliases") or []
+    if raw_id:
+        aliases.append(raw_id)
+        aliases = dedupe(original=aliases)
+    date_published = get_published_date(raw_data=raw_data)
+    severities = list(get_severities(raw_data=raw_data))
+    references = get_references(raw_data=raw_data, severities=severities)
+
+    affected_packages = []
+
+    for affected_pkg in raw_data.get("affected") or []:
+        purl = get_affected_purl(affected_pkg=affected_pkg, raw_id=raw_id)
+        ranges = affected_packages.get("ranges") or []
+        events = ranges[0].get("events") or []
+        if not purl:
+            logger.error(f"Unsupported package type: {affected_pkg!r} in OSV: {raw_id!r}")
+            continue
+
+        affected_version_range = get_affected_version_range(
+            affected_pkg=affected_pkg,
+            raw_id=raw_id,
+            supported_ecosystem=purl.type,
+        )
+
+        for fixed_range in affected_pkg.get("ranges") or []:
+            fixed_version = get_fixed_versions(
+                fixed_range=fixed_range,
+                raw_id=raw_id,
+                supported_ecosystem=purl.type,  # can use these information in future to update the get_fixed_version function.
+            )
+
+            for version in fixed_version:
+                affected_packages.append(
+                    AffectedPackage(
+                        package=purl,
+                        affected_version_range=affected_version_range,
+                        fixed_version=version,
+                    )
+                )
+
+    database_specific = raw_data.get("database_specific") or {}
+    cwe_ids = database_specific.get("cwe_ids") or []
+    weaknesses = list(map(get_cwe_id, cwe_ids))
+
+    return AdvisoryData(
+        aliases=aliases,
+        summary=summary,
+        references=references,
+        affected_packages=affected_packages,
+        date_published=date_published,
+        weaknesses=weaknesses,
+        url=advisory_url,
+    )
+
+
+def get_affected_purl(affected_pkg, raw_id):
+    package = affected_pkg.get("package") or {}
+    purl = package.get("purl")
+    if purl:
+        try:
+            purl = PackageURL.from_string(purl)
+        except ValueError:
+            logger.error(
+                f"Invalid PackageURL: {purl!r} for OSV "
+                f"affected_pkg {affected_pkg} and id: {raw_id}"
+            )
+
+    else:
+        ecosys = package.get("ecosystem")
+        name = package.get("name")
+        purl = PackageURL(type="rpm", namespace="almalinux", name=name)
+
+    return PackageURL.from_string(str(purl))
+
+
+def get_fixed_versions(fixed_range) -> List[Version]:
+    fixed_versions = []
+    fixed_range_type = fixed_range["type"]
+    for version in extract_fixed_versions(fixed_range):
+        fixed_versions.append(RpmVersion(version))
+    return dedupe(fixed_versions)
+
+
+def get_affected_version_range(affected_pkg, raw_id, supported_ecosystem):
```

Comment:
@ziadhany unable to figure out how to find affected version range

Reply:
@ambuj-1211, we don't handle getting the [affected version range](https://ossf.github.io/osv-schema/#affected-fields) from `introduced` and `fixed` in OSV. We only obtain this using [versions](https://ossf.github.io/osv-schema/#affectedversions-field).

I recommend handling this in a separate issue. Please create one for it.

Additionally, please review the following link:
https://github.com/nexB/vulnerablecode/blob/main/vulnerabilities/tests/test_osv.py#L270

If we don't have a affected [versions](https://ossf.github.io/osv-schema/#affectedversions-field) set this as null.

Reply:
@ziadhany I have done some changes, let me know if it require something else and also tell me if I need to squash the commits to make the history cleaner.